### PR TITLE
[dashboard] BlockedRepo UI

### PR DESCRIPTION
**Pull request by geropl**
## Description

This still looks... ugly, but fixes the general issue of not being able to add blocked repositories from the UI.

image

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [x] /werft with-preview

### CODE REVIEW

**Comment by geropl** [comment]
@AlexTugarev Can you maybe 10mins tomorrow and help me fix the basics for the :skateboard: ? :pray: 

@gtsiolis I guess you're pretty busy, but: It would be awesome if you could have a look (independent of when we merge this) and come up with a nice version - happy to implement that. :sparkles:

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Review by gtsiolis — APPROVED** [review]
Thanks for adding this, @geropl! 🔮 

Left some comments in case we'd like to address all these UX issues here or in a separate PR.

Approving to unblock merging this, but holding to let someone from the `@gitpod-io/engineering-webapp` to take a closer look at the code changes. 🏓 

/hold

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (andrew-farries) [inline components/dashboard/src/admin/BlockedRepositories.tsx:26]: Can we remove this empty interface?

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (andrew-farries) [inline components/dashboard/src/admin/BlockedRepositories.tsx:36]: Can you do useState<BlockedRepository> and get rid of the need to cast the argument?

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (andrew-farries) [inline components/dashboard/src/admin/BlockedRepositories.tsx:66]: Same question about using a type when using the `useState` hook to get rid of the `as` here?

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (andrew-farries) [inline components/dashboard/src/admin/BlockedRepositories.tsx:186]: What is `pev`? Should it be `prev` for `previous`?

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (andrew-farries) [inline components/dashboard/src/admin/BlockedRepositories.tsx:184]: How come we need the `useRef` hook here? It' not something I've seen used much before?

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:36]: Yes, make sense. Introduced the type `ExistingBlockedRepository` to make that work. :+1:

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:66]: Same as above.

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:184]: Because we don't have access to the state directly here, but don't want to manipulate the part of the state that is displayed, but just a version of it.
Maybe this can solves differently and easier. This is what I saw in the EnvVarSettings. :+1:

**Review by andrew-farries — APPROVED** [review]

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:153]: It's trivial, but as I want to ship very soon, and we don't have the API just yet I skipped it in this :skateboard: :lotus_position:

Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:153]: Ah, thank you for the nudge: Cleaner _and_ easier! :orange_heart:

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:142]: I added same basic validation for an empty regex. But neither cases are break functionality.

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:288]: I didn't go with the "Future", because I find that confusing. But happy to learn about the motivation and be convinced otherwise! :thinking:

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx:177]: I absolutely agree, but could find an example where I could ~copy from~ get inspiration from. :roll_eyes:

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (geropl) [reply components/dashboard/src/admin/BlockedRepositories.tsx]: Let's do that in a follow-up :+1:

**Comment by geropl** [comment]
/unhold

I addresses as much of the suggestions as possible to still get the release train :train2:  :laughing:

**Comment by geropl** [comment]
Thank you @andrew-farries and @gtsiolis for your valuable feedback! :pray:

File: components/dashboard/src/admin/BlockedRepositories.tsx
Comment (gtsiolis) [reply components/dashboard/src/admin/BlockedRepositories.tsx:137]: @geropl The linked issues is about the misuse of the alert component below the search input, which is misaligned and using poor copy sometimes.

**Pull request merged**
